File: java/src/main/java/challenges/aoc2024/day8/Application.java
package challenges.aoc2024.day8;

import challenges.aoc2024.day8.service.AntennaService;
import challenges.aoc2024.day8.service.DayEightSolver;
import service.TaskSolver;

public class Application {
    public static void main(String[] args) {
        final TaskSolver solver = new TaskSolver(new DayEightSolver(new AntennaService()));
        solver.solveTasks();
    }
}

File: scala/src/main/scala/challenges/aoc2025/day3/Application.scala
package challenges.aoc2025.day3

import _root_.service.TaskSolver
import challenges.aoc2025.day3.service.{BatteryService, DayThreeSolver}

object Application {
  def main(args: Array[String]): Unit = {
    val solver: TaskSolver = TaskSolver(DayThreeSolver(BatteryService()))
    solver.solveTasks()
  }
}
